README.md

```diff
@@ -22,3 +22,35 @@ or using a browser: `http://localhost:9090`
 
 Please refer to the [lorax-composer](https://github.com/weldr/lorax)'s documenation as osbuild-composer is a drop-in replacement.
 
+## High-level overview
+
+![overview](docs/osbuild-composer.svg)
+
+### Frontends
+
+`osbuild-composer` is meant to be used with 3 different front-ends. The primary one, which is meant for general use, is cockpit-composer. It is part of the Cockpit project and unless you have a strong reason not to use it, you should use it. `composer-cli` is a command line tool that can be used for non-interactive usage. Finally, RCM client (TBD) is a special-purpose client that is not meant for general usage.
+
+### Compose
+* Compose is what the user submits over one of the frontends
+* It contains of one or more **image builds**
+* It contains zero or more **upload actions**
+
+### Image build
+* The resulting *image* has a *type*: https://github.com/osbuild/osbuild-composer/blob/master/internal/distro/fedora30/distro.go#L19
```

Comment:
That link doesn't tell me anything about supported types. Maybe it should reference `composer-cli compose types` instead?

Reply:
I think about this README file as a file for developers, but it is worth mentioning.

Reply:
Even as a developer that line didn't help. I see that variable is populated later with the types but it's not immediately clear when looking at that line.

Reply:
Sure, but I still think it is a pointer in the right direction. There is a whole section about "image type" below this one.